src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -187,15 +289,103 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
     /* ------------------------------------------------------------------------ */
     provname = MPIR_CVAR_OFI_USE_PROVIDER ? (char *) MPL_strdup(MPIR_CVAR_OFI_USE_PROVIDER) : NULL;
     hints->fabric_attr->prov_name = provname;
+
+    /* If the user picked a particular provider, ignore the checks */
+    if (MPIDI_OFI_ENABLE_RUNTIME_CHECKS) {
```

Comment:
I originally had this as preprocessor checks and @charlesarcher had some...disagreements...

The assertion is that these all end up getting compiled out because `MPIDI_OFI_ENABLE_RUNTIME_CHECKS` is a compile time constant so the compiler can optimize it out easily. By having this as "runtime checks", it makes things more readable without actually adding a branch.